src/conn/conn_compact.c

```diff
@@ -18,13 +20,233 @@ __compact_server_run_chk(WT_SESSION_IMPL *session)
     return (FLD_ISSET(S2C(session)->server_flags, WT_CONN_SERVER_COMPACT));
 }
 
+/*
+ * __background_compact_list_insert --
+ *     Insert compaction statistics for a file into the background compact list.
+ */
+static void
+__background_compact_list_insert(WT_SESSION_IMPL *session, WT_BACKGROUND_COMPACT_STAT *compact_stat)
+{
+    WT_CONNECTION_IMPL *conn;
+    uint64_t bucket, hash;
+
+    conn = S2C(session);
+
+    hash = __wt_hash_city64(compact_stat->uri, strlen(compact_stat->uri));
+    bucket = hash & (conn->hash_size - 1);
+
+    TAILQ_INSERT_HEAD(&(conn)->background_compact.compactqh, compact_stat, q);
+    TAILQ_INSERT_HEAD(&(conn)->background_compact.compacthash[bucket], compact_stat, hashq);
+    ++(conn)->background_compact.file_count;
+}
+
+/*
+ * __background_compact_list_remove --
+ *     Remove compaction statistics for a file from the background compact list.
+ */
+static void
+__background_compact_list_remove(WT_SESSION_IMPL *session, WT_BACKGROUND_COMPACT_STAT *compact_stat)
+{
+    WT_CONNECTION_IMPL *conn;
+    uint64_t bucket, hash;
+
+    conn = S2C(session);
+
+    hash = __wt_hash_city64(compact_stat->uri, strlen(compact_stat->uri));
+    bucket = hash & (conn->hash_size - 1);
+
+    TAILQ_REMOVE(&(conn)->background_compact.compactqh, compact_stat, q);
+    TAILQ_REMOVE(&(conn)->background_compact.compacthash[bucket], compact_stat, hashq);
+    --(conn)->background_compact.file_count;
```

Comment:
Can we assert we are always greater than 0 before decrementing?

Reply:
Fixed https://github.com/wiredtiger/wiredtiger/pull/9541/commits/138801431ace7c1d77267584a57f12d92db03d2f.